ui/src/app/machines/views/MachineDetails/MachineConfiguration/TagForm/TagForm.tsx

```diff
@@ -45,55 +46,36 @@ const TagForm = ({ systemId }: Props): JSX.Element | null => {
   }
 
   return (
-    <Row>
-      <Col size={3}>
-        <div className="u-flex--between u-flex--wrap">
-          <h4>Tags</h4>
-          {canEdit && !editing && (
-            <Button
-              className="u-no-margin--bottom u-hide--large"
-              onClick={() => setEditing(true)}
-            >
-              Edit
-            </Button>
-          )}
-        </div>
-      </Col>
-      <Col size={editing ? 9 : 6}>
-        {editing ? (
-          <TagActionForm
-            clearHeaderContent={() => setEditing(false)}
-            errors={errors}
-            machines={[machine]}
-            processingCount={taggingMachines.length}
-            viewingDetails
-            viewingMachineConfig
+    <EditableSection
+      canEdit={canEdit}
+      editing={editing}
+      hasSidebarTitle
+      setEditing={setEditing}
+      title="Tags"
+    >
+      {editing ? (
```

Comment:
Did you consider moving the editing state inside the component? It seems like it might be nice to to have to manage the state in the parent component. Maybe something like this:

<EditableSection
      form={<TagActionForm ...}
    >
    <p>
      ...